gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -248,7 +253,7 @@ public void orchestrate(Spec spec, Properties jobProps, long triggerTimestampMil
         Instrumented.markMeter(this.flowOrchestrationFailedMeter);
         return;
       }
-      Map<String, String> flowMetadata = TimingEventUtils.getFlowMetadata((FlowSpec) spec);
+      Map<String, String> flowMetadata = TimingEventUtils.getFlowMetadata(flowSpec);
```

Comment:
much nicer to get rid of those casts so far separated from the `instanceof` check!